### PR TITLE
Exposing batch job resource-v2

### PR DESCRIPTION
**Pull request by TajiHarrisMicrosoft**
Expose batch job resources to sdk

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

Creating curated entity to allow sdk user to be exposed to Batch Job Resource class

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_deployment/batch/batch_job_property.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    path = fields.Str()
+
+    @post_load
+    def make(self, data: Any, **kwargs: Any) -> Any:
+        from azure.ai.ml.entities._deployment.data_version import DataVersion
+
+        return DataVersion(**data)
+
+
+class BatchJobPropertySchema(PathAwareSchema):
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_deployment/batch/batch_job_property.py:38]: Multiple fields of BatchJobPropertySchema already exist in BatchJobSchema. Why the duplication? Can we get rid of the `properties` nesting of which I don't see much customer value?

Comment (TajiHarrisMicrosoft) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_deployment/batch/batch_job_property.py:38]: In the design it was agreed upon that we would expose all the values in batch job property at the level of batch job. However when invoke is called it returns a batch job property that we must use to create the batch job object that get returned to the cli. However, I can remove the properties attribute.

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_deployment/batch/batch_job.py [hunk 2]
```diff
@@ -2,68 +2,46 @@
...
+    name = fields.Str()
+    type = fields.Str()
+    system_data = NestedField(SystemDataSchema)
     compute = NestedField(ComputeBindingSchema)
     dataset = fields.Str()
     error_threshold = fields.Int()
     input_data = fields.Dict()
     mini_batch_size = fields.Int()
-    name = fields.Str(data_key="job_name")
+    batch_job_name = fields.Str()
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_deployment/batch/batch_job.py:33]: What's the difference between name and batch_job_name?

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_deployment/batch/batch_job_property.py (hunk 1 above)
Comment (luigiw) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_deployment/batch/batch_job_property.py:38]: I'm not following why batch job property has to be returned to cli?

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_batch_endpoint_operations.py [hunk 3]
```diff
@@ -94,6 +95,16 @@ def __init__(
     def _datastore_operations(self) -> "DatastoreOperations":
         return self._all_operations.all_operations[AzureMLResourceType.DATASTORE]
 
+    @property
+    def _dataset_dataplane_client(self) -> ServiceClientDataPlanePreview:
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_batch_endpoint_operations.py:99]: Minor: why is this called "dataset_dataplane_client"?

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/data_version.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+from typing import Dict
+from azure.ai.ml._schema._deployment.batch.batch_job_property import OutputDataSchema
+from azure.ai.ml._schema.core.fields import ArmStr
+from azure.ai.ml.constants._common import BASE_PATH_CONTEXT_KEY
+
+
+class DataVersion:
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/data_version.py:11]: DataVersion and ComputeBinding are both legacy concepts. I'm hessitant to expose these to customers. What's the plan to refactor BatchJob to match other job structures?

**Pull request closed without merging**